### PR TITLE
Adds preferDeals support

### PR DESCRIPTION
**Pull request by hhhjort**
Solves #1355

### CODE REVIEW

File: exchange/auction.go [hunk 1]
```diff
@@ -125,6 +125,19 @@ func newAuction(seatBids map[openrtb_ext.BidderName]*pbsOrtbSeatBid, numImps int
 	}
 }
 
+// Calculate if the new bid (nbid) will win against the current winning bid (wbid) given preferDeals.
```
Comment (SyntaxNode) [inline exchange/auction.go:128]: Nitick: Per Go's comment convention, should be something like:
```
isNewWinningBid calculates if the new bid (nbid) will win against the current winning bid (wbid) given preferDeals.
```

File: exchange/auction.go [hunk 2]
```diff
@@ -125,6 +125,19 @@ func newAuction(seatBids map[openrtb_ext.BidderName]*pbsOrtbSeatBid, numImps int
...
+func isNewWinningBid(nbid, wbid *openrtb.Bid, preferDeals bool) bool {
```
Comment (SyntaxNode) [inline exchange/auction.go:129]: Consider changing the variable names to be more obvious:
nbid -> bid (i think the 'new' part is implied)
wbid -> winningBid

File: exchange/auction_test.go [hunk 3]
```diff
@@ -280,6 +280,227 @@ func runCacheSpec(t *testing.T, fileDisplayName string, specData *cacheSpec) {
 	}
 }
 
+type testNewAuctionSpec struct {
```
Comment (SyntaxNode) [inline exchange/auction_test.go:283]: This type has no use outside of the `TestNewAuction` method. Why not use an anonymous struct?
```
testCases := []struct {
	description     string
	seatBids        map[openrtb_ext.BidderName]*pbsOrtbSeatBid
	numImps         int
	preferDeals     bool
	expectedAuction auction,
}{
	{
	...
	},
}
```

**Review by SyntaxNode — DISMISSED** [review]

**Review by SyntaxNode — APPROVED** [review]

**Review by bsardo — APPROVED** [review]
LGTM

**Pull request merged**